dcc-portal-ui/app/scripts/advanced/js/advanced.js

```diff
@@ -65,8 +65,34 @@ angular.module('icgc.advanced.controllers', [
 
       _locationFilterCache = _filterService.getCachedFiltersFactory();
 
-      var _isInAdvancedSearchCtrl = true;
+      _controller.donorSets = _.cloneDeep(SetService.getAllDonorSets());
+      _controller.geneSets = _.cloneDeep(SetService.getAllGeneSets());
+      _controller.mutationSets = _.cloneDeep(SetService.getAllMutationSets());
+
+      // to check if a set was previously selected and if its still in effect
+      const checkSetInFilter = (entity, entitySets) => {
+        let filters = _locationFilterCache.filters();
+        const setIdPrefix = 'ES:';
+
+        if(filters[entity] && filters[entity].id){
+          _.each(entitySets, (set) => {
+            set.selected = false;
+            _.each(filters[entity].id.is, (id) => {
+              if(_.includes(id, setIdPrefix)){
+                if(`${setIdPrefix}${set.id}` === id){
+                  set.selected = true;
+                }
+              }
+            });
+          });
+        } else {
+          _.each(entitySets, (set) => {
+            set.selected = false;
+          });
+        }
+      };
```

Comment:
Aah, I took mutation as an actual mutation that happens in genes. Anyway, I don't think `get` method implies that it returns a Boolean. It could return an Object or a string. i.e `getAllDonorSets()`. Same with `check`. The method simply checks if the set exist in the applied filters. If you have better name that justifies the action then by all means. I am open to suggestion.